src/models/room.ts

```diff
@@ -1616,8 +1679,16 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
         return this.setUnreadNotificationCount(type, count);
     }
 
-    public setSummary(summary: IRoomSummary): void {
-        const heroes = summary["m.heroes"];
+    public setSummary(summary: IRoomSummary | IRoomSummaryMSC4186): void {
+        // map heroes onto the MSC4186 form as that has more data
+        const heroes = summary["m.heroes"]?.map((h) => {
+            if (typeof h === "string") {
+                return {
+                    userId: h,
+                };
+            }
+            return h;
+        });
```

Comment:
mild suggestion for clarity

```suggestion
        const heroes = summary["m.heroes"]?.map(
            (h) => (typeof h === "string") ? { userId: h } : h
        );
```